watertap/flowsheets/dye_desalination/dye_desalination.py

```diff
@@ -895,29 +970,57 @@ def total_externalities(b):
         if hasattr(m.fs, "pretreatment"):
             return pyunits.convert(
                 m.fs.water_recovery_revenue
-                - m.fs.dye_disposal_cost
+                - m.fs.dye_cost
                 - m.fs.brine_cost
                 - m.fs.sludge_disposal_cost,
                 to_units=pyunits.USD_2020 / pyunits.year,
             )
         else:
             return pyunits.convert(
-                m.fs.water_recovery_revenue - m.fs.dye_disposal_cost - m.fs.brine_cost,
+                m.fs.water_recovery_revenue - m.fs.dye_cost - m.fs.brine_cost,
                 to_units=pyunits.USD_2020 / pyunits.year,
             )
 
     @m.fs.Expression(doc="Total cost of brine/sludge/dye disposed")
     def external_cost(b):
         if hasattr(m.fs, "pretreatment"):
             return pyunits.convert(
-                m.fs.dye_disposal_cost + m.fs.brine_cost + m.fs.sludge_disposal_cost,
+                m.fs.dye_cost + m.fs.brine_cost + m.fs.sludge_disposal_cost,
                 to_units=pyunits.USD_2020 / pyunits.year,
             )
         else:
-            return pyunits.convert(
-                m.fs.dye_disposal_cost + m.fs.brine_cost,
-                to_units=pyunits.USD_2020 / pyunits.year,
-            )
+            if dye_revenue and brine_revenue:
+                return pyunits.convert(
+                    0 * pyunits.USD_2020 / pyunits.year,
+                    to_units=pyunits.USD_2020 / pyunits.year,
+                )
+            elif dye_revenue:
+                if value(m.fs.brine_cost) > 0:
+                    return pyunits.convert(
+                        m.fs.brine_cost,
+                        to_units=pyunits.USD_2020 / pyunits.year,
```

Comment:
Something to keep in mind for subsequent PR and report modifications: would be good to move away from 2020 currency since it's more of an anomaly.

Reply:
@adam-a-a So what should be used instead?

Reply:
Could go with 2023